### PR TITLE
Add Code coverage

### PR DESCRIPTION
**Pull request by GregoireHebert**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~
| License       | MIT

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,43 +5,108 @@ on:
...
       - name: Run php-cs-fixer tests
         run: vendor/bin/php-cs-fixer fix --dry-run --diff
+
+      - name: Enable code coverage
+        if: matrix.coverage && matrix.dependency == ''
+        run: echo '::set-env name=COVERAGE::1'
+
+      - name: Run PHPUnit tests
+        run: |
+          mkdir -p build/logs/phpunit
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]: Isn't it possible to split this test in 1 tests, using `if` syntax?:
```
- name: Run PHPUnit tests with coverage
  if: matrix.coverage
  run: vendor/bin/phpunit --coverage-clover ...

- name: Run PHPUnit tests
  if: !matrix.coverage
  run: vendor/bin/phpunit ...
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -5,43 +5,108 @@ on:
...
+
       - name: Run phpstan tests lowest
         if: matrix.dependency != ''
         run: vendor/bin/phpstan analyze -c phpstan.neon.lowest.dist
+
       - name: Run php-cs-fixer tests
         run: vendor/bin/php-cs-fixer fix --dry-run --diff
+
+      - name: Enable code coverage
+        if: matrix.coverage && matrix.dependency == ''
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]:
```suggestion
        if: matrix.coverage
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,43 +5,108 @@ on:
...
+        run: |
+          mkdir -p build/logs/phpunit
+          if [ "$COVERAGE" = '1' ]; then
+            vendor/bin/phpunit --coverage-clover build/logs/phpunit/clover.xml --colors=always
+          else
+            vendor/bin/phpunit --colors=always
+          fi
+
+      - name: Upload test artifacts
+        if: matrix.coverage && matrix.dependency == ''
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]:
```suggestion
        if: matrix.coverage
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -5,43 +5,108 @@ on:
...
+        uses: actions/upload-artifact@v1
+        with:
+          name: phpunit-logs-php${{ matrix.php }}
+          path: build/logs/phpunit
+
+      - name: Upload coverage results to Codecov
+        if: matrix.coverage && matrix.dependency == ''
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]:
```suggestion
        if: matrix.coverage
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -5,43 +5,108 @@ on:
...
+        uses: codecov/codecov-action@v1.0.3
+        with:
+          name: phpunit-php${{ matrix.php }}
+          flags: phpunit
+          token: ${{ secrets.CODECOV_TOKEN }}
+
+      - name: Upload coverage results to Coveralls
+        if: matrix.coverage && matrix.dependency == ''
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml]:
```suggestion
        if: matrix.coverage
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -5,43 +5,108 @@ on:
...
+  COVERAGE: '0'
+  EXT_PCOV_VERSION: '1.0.6'
+
 jobs:
   tests:
     name: Tests (PHP ${{ matrix.php }})
     runs-on: ubuntu-latest
     container:
-      image: php:${{ matrix.php }}
+      image: php:${{ matrix.php }}-alpine
```
[the remark this answers is not in the document]
Comment (GregoireHebert) [reply .github/workflows/ci.yml]: yeah I saw... :/

File: .github/workflows/ci.yml (hunk 2 above)
Comment (GregoireHebert) [reply .github/workflows/ci.yml]: no need to run the coverage for each matrices. the coverage won't change.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (GregoireHebert) [reply .github/workflows/ci.yml]: I'll try something.

File: .github/workflows/ci.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (GregoireHebert) [reply .github/workflows/ci.yml]: cool, made it work with standard php image :)

**Pull request merged**